libc/CMakeLists.txt

```diff
@@ -46,7 +46,11 @@ set(LIBC_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
 
 set(LIBC_ENABLE_USE_BY_CLANG OFF CACHE BOOL "Whether or not to place libc in a build directory findable by a just built clang")
 
-set(LIBC_KERNEL_HEADERS "/usr/include" CACHE STRING "Path to Linux kernel headers")
+# LLVM_LIBC_USE_HOST_KERNEL_HEADERS is brittle, and frowned upon (prefer
+# setting LIBC_KERNEL_HEADERS), but is simpler from a hello world linux host
+# build.
+set(LLVM_LIBC_USE_HOST_KERNEL_HEADERS OFF CACHE BOOL "Add /usr/include to the header seach path")
+set(LIBC_KERNEL_HEADERS "" CACHE STRING "Path to Linux kernel headers")
```

Comment:
oh, I wonder if we should be using `option` instead? I see that used elsewhere in our cmake. I'm not familiar with `option`...

Reply:
There are two ways to define a boolean flag in CMake:
```
set(FOO true CACHE BOOL "description")
option(FOO "description" ON)
```
The difference between the two is that the `option` command does _not_ create a `CACHE` variable if a normal variable with the same name exists, but using `set` with `CACHE` variable named `FOO` when `FOO` exists as a normal variable will yield two copies of `FOO`, a normal variable and a `CACHE` one.

Which is one more appropriate really depends on your intention. There are places in LLVM build system where we intentionally rely on the `set` semantics (e.g. we would use the `CACHE` variable as the default value to initialize the normal variable).

Reply:
Ok, so it sounds like cmake has distinct "namespace-esque" differentiation between "normal variables" and "cache variables."

For cmake variables that we encourage folks to use to configure llvm-libc, I _think_ we may want `option`. That seems very common in `libcxx/CMakeLists.txt`.

Reply:
76e422c57e63fc24e065c1ab74a49a45afaaac84